vtr_flow/arch/COFFE_22nm/agilex_arch.xml

```diff
@@ -0,0 +1,2114 @@
+          <T_clock_to_Q max="60.32e-12" port="mem_1024x20_sp.data" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_1024x20_sp.we" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_1024x20_sp.out" clock="clk"/>
+          <power method="pin-toggle">
+            <port name="clk" energy_per_toggle="9.0e-12"/>
+            <static_power power_per_instance="0.0"/>
+          </power>
+        </pb_type>
+        <interconnect>
+          <direct name="address1" input="memory.addr1[9:0]" output="mem_1024x20_sp.addr">
+            <delay_constant max="0.973e-9" in_port="memory.addr1[9:0]" out_port="mem_1024x20_sp.addr"/>
+          </direct>
+          <direct name="data1" input="memory.data[19:0]" output="mem_1024x20_sp.data">
+            <delay_constant max="0.973e-9" in_port="memory.data[19:0]" out_port="mem_1024x20_sp.data"/>
+          </direct>
+          <direct name="writeen1" input="memory.we1" output="mem_1024x20_sp.we">
+            <delay_constant max="0.973e-9" in_port="memory.we1" out_port="mem_1024x20_sp.we"/>
+          </direct>
+          <direct name="dataout1" input="mem_1024x20_sp.out" output="memory.out[19:0]">
+            <delay_constant max="0.973e-9" in_port="mem_1024x20_sp.out" out_port="memory.out[19:0]"/>
+          </direct>
+          <direct name="clk" input="memory.clk" output="mem_1024x20_sp.clk">
+             </direct>
+        </interconnect>
+      </mode>
+
+      <mode name="mem_2048x10_sp">
+        <pb_type name="mem_2048x10_sp" blif_model=".subckt single_port_ram" class="memory" num_pb="1">
+          <input name="addr" num_pins="11" port_class="address"/>
+          <input name="data" num_pins="10" port_class="data_in"/>
+          <input name="we" num_pins="1" port_class="write_en"/>
+          <output name="out" num_pins="10" port_class="data_out"/>
+          <clock name="clk" num_pins="1" port_class="clock"/>
+          <T_setup value="51.12e-12" port="mem_2048x10_sp.addr" clock="clk"/>
+          <T_setup value="51.12e-12" port="mem_2048x10_sp.data" clock="clk"/>
+          <T_setup value="51.12e-12" port="mem_2048x10_sp.we" clock="clk"/>
+          <T_setup value="18.91e-12" port="mem_2048x10_sp.out" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_2048x10_sp.addr" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_2048x10_sp.data" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_2048x10_sp.we" clock="clk"/>
+          <T_clock_to_Q max="60.32e-12" port="mem_2048x10_sp.out" clock="clk"/>
+          <power method="pin-toggle">
+            <port name="clk" energy_per_toggle="9.0e-12"/>
+            <static_power power_per_instance="0.0"/>
+          </power>
+        </pb_type>
+        <interconnect>
+          <direct name="address1" input="memory.addr1[10:0]" output="mem_2048x10_sp.addr">
+            <delay_constant max="0.973e-9" in_port="memory.addr1[10:0]" out_port="mem_2048x10_sp.addr"/>
```

Comment:
This timing specification looks strange to me.  See https://docs.verilogtorouting.org/en/latest/tutorials/arch/timing_modeling/#arch-model-timing-tutorial (the sequential RAM example).  
You seem to be specifying the RAM delay (read path) on the interconnect from the cluster level RAM inputs to the internal primitive inputs, and similarly for the outputs. 
Instead I think you should be specifying the RAM (read path) delay from the primitive RAM block inputs to the primitive RAM block outputs (that is the internals of the RAM).  (e.g. mem_2048x10_sp.addr -> mem_2048x10_sp.out).

Reply:
Oops. I added the delay_constant at the wrong place. Fixing now.